### PR TITLE
Created agreement cost and count queries for 2G

### PR DESCRIPTION
**Pull request by adoerrer**
Adresses #282 and #283

Can only be tested against `reports_dev` 
Be aware that two derived tables have updates in this PR too and can be exchanged in `erm_agreement_count.sql` by `axeldoerrer_agreements_erm_resource`  and `axeldoerrer_agreements_subscription_agreement_entitlement ` in order to run it.

### CODE REVIEW

**Review by stdombek — APPROVED** [review]

**Review by amzoss — CHANGES_REQUESTED** [review]
I think I found a mistake in the join for the erm_agreement_count query, but otherwise looks good!

File: sql_2g/report_queries/erm/count/erm_agreement_count.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+*/
+SELECT
+    sa_ent.subscription_agreement_name AS "Agreements",
+    agrestat.rdv_label AS "Status",
+    count(DISTINCT pci_list.pci_id) AS "Count"
+FROM
+    folio_reporting.agreements_package_content_item AS pci_list
+    LEFT JOIN folio_reporting.agreements_subscription_agreement_entitlement AS sa_ent ON pci_list.entitlement_id = sa_ent.entitlement_id
+    LEFT JOIN folio_agreements.refdata_value AS agrestat ON agrestat.rdv_id = sa_ent.subscription_agreement_agreement_status
```
[the remark this answers is not in the document]
Comment (stdombek) [reply sql_2g/report_queries/erm/count/erm_agreement_count.sql]: The attribute for the status in the derived table, folio_reporting.agreements_subscription_agreement_entitlement, is directly called sa_ent.subscription_agreement_agreement_status with twice agreement in the attribute name. Should that be changed?

**Review by stdombek — APPROVED** [review]
The status attribute in the derived table, folio_reporting.agreements_subscription_agreement_entitlement, is changed from subscription_agreement_agreement_status to subscription_agreement_status. Then everything should be right.

File: sql_2g/report_queries/erm/count/erm_agreement_count.sql [hunk 2]
```diff
@@ -8,16 +8,67 @@ TABLES
...
+		-- filters on erm_resource level
+		''::VARCHAR AS resource_type, -- Enter your erm resource type eg. 'monograph', 'serial' etc.
+		-- filters on entitelment level
+        NULL::DATE AS ent_start_date,
+        NULL::DATE AS ent_end_date,
+        --'2021-01-01' :: DATE AS ent_start_date, -- start date day is included in interval
+        --'2022-01-01' :: DATE AS ent_end_date, -- end date day is NOT included in interval -> enter next day	
+		-- filters on package content item level
+        NULL::DATE AS pci_start_date,
+        NULL::DATE AS pci_end_date,
```
Comment (amzoss) [inline sql_2g/report_queries/erm/count/erm_agreement_count.sql:31]: extra comma at the end of line should be removed

File: sql_2g/report_queries/erm/count/erm_agreement_count.sql [hunk 3]
```diff
@@ -8,16 +8,67 @@ TABLES
...
+	    (pci_list.pci_access_end >= (SELECT pci_start_date FROM parameters) OR pci_list.pci_access_end IS NULL))
+	    OR 
+		(((SELECT pci_start_date FROM parameters) IS NULL) 
+			OR ((SELECT pci_end_date FROM parameters) IS NULL)))
+	AND 			
+    ((sa_ent.entitlement_active_from < (SELECT ent_end_date FROM parameters) AND
+	    (sa_ent.entitlement_active_to >= (SELECT ent_start_date FROM parameters) OR sa_ent.entitlement_active_to IS NULL))
+	    OR 
+		(((SELECT ent_start_date FROM parameters) IS NULL) 
+			OR ((SELECT ent_end_date FROM parameters) IS NULL)))
```
Comment (amzoss) [inline sql_2g/report_queries/erm/count/erm_agreement_count.sql:59]: These did not work for me. In metadb, the `pci_list.pci_access_start`, `pci_list.pci_access_end`, `sa_ent.entitlement_active_from`, and `sa_ent.entitlement_active_to` fields all seem to be integers, so I get the following error when trying to run:

`operator does not exist: integer < date`

**Review by amzoss — CHANGES_REQUESTED** [review]

File: sql_2g/report_queries/erm/count/erm_agreement_count.sql (hunk 3 above)
Comment (stdombek) [reply sql_2g/report_queries/erm/count/erm_agreement_count.sql:59]: At the moment these are integers. Nassib is working on making Date fields. This is something that has yet to be resolved in metadb.

File: sql_2g/report_queries/erm/count/erm_agreement_count.sql (hunk 2 above)
Comment (adoerrer) [reply sql_2g/report_queries/erm/count/erm_agreement_count.sql]: Thanks, have it removed

File: sql_2g/report_queries/erm/count/erm_agreement_count.sql (hunk 3 above)
Comment (adoerrer) [reply sql_2g/report_queries/erm/count/erm_agreement_count.sql:59]: Yes, I forgot to mention. Thanks Stefan for bringing this up!

**Review by amzoss — APPROVED** [review]
Thanks, looks good!

**Pull request merged**